overlord/snapstate/handlers.go

```diff
@@ -1926,24 +1925,31 @@ func setMigrationFlagsinState(snapst *SnapState, snapsup *SnapSetup) {
 	}
 }
 
-// maybeRestart will schedule a reboot or restart as needed for the
-// just linked snap with info if it's a core or snapd or kernel snap.
-func (m *SnapManager) maybeRestart(t *state.Task, info *snap.Info, rebootInfo boot.RebootInfo) {
+// maybeRestart carries information to decide whether a restart of some form
+// is required.
+type maybeRestart struct {
+	info *snap.Info
+	boot.RebootInfo
+}
+
+// finishTaskWithMaybeRestart will set the final status for the task
+// and schedule a reboot or restart as needed for the just linked snap
+// with info if it's a core, snapd or kernel snap.
```

Comment:
maybe
```suggestion
// and schedule a reboot or restart as needed for the just linked snap
// passed in the {maybeRestart,restartInfo} parameter.
```
That we use the snap's type to take a decision seems an implementation detail to me, but if you think it's important you can add "...based on its type".